src/wp-includes/blocks.php

```diff
@@ -425,6 +425,21 @@ function unregister_block_type( $name ) {
 function has_blocks( $post = null ) {
 	if ( ! is_string( $post ) ) {
 		$wp_post = get_post( $post );
+
+		if ( null === $wp_post ) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %s: The $post variable. */
+					__( '%s is not a valid post.' ),
+					'<code>$post</code>'
+				),
+				'6.1.0'
+			);
```

Comment:
👋 As an observer, `_doing_it_wrong()` seems a little strong. What's the thinking behind it? Most other core functions would just `return false` (or whatever appropriate value) so triggering a notice here seems inconsistent. Also, what's "wrong" with this case? The post might not exist, but is the function being used incorrectly?

Reply:
Hi @dlh01, after an impromptu chat with @peterwilsoncc and reviewing other instances in Core, I believe you're absolutely right! Thanks! I'll make the changes to the PR now.